lib/active_model/serializer/concerns/associations.rb

```diff
@@ -83,15 +83,17 @@ def associate(reflection)
       #   +default_include_directive+ config value when not provided)
       # @return [Enumerator<Association>]
       #
-      def associations(include_directive = ActiveModelSerializers.default_include_directive)
+      def associations(include_directive = ActiveModelSerializers.default_include_directive, include_slice = nil)
+        include_slice ||= include_directive
         return unless object
 
         Enumerator.new do |y|
           self.class._reflections.values.each do |reflection|
             next if reflection.excluded?(self)
             key = reflection.options.fetch(:key, reflection.name)
             next unless include_directive.key?(key)
```

Comment:
So in this case, for instance, let's say your include directive was:

```ruby
{ posts: { comments: {} } }
```

When you are iterating the relations and you hit the `comments` portion, the `include_directive` (the current code) is actually '*', assuming no sparse fieldsets requested. But the `include_slice` would be {}.